### PR TITLE
CB-16541 Enable scaling when the cluster is in NODE_FAILURE state, ad…

### PR DESCRIPTION
**Pull request by bbihari**
…d e3e test

The first scale up operation will start from NODE_FAILURE state.

### CODE REVIEW

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCloudFormationSetup.java [hunk 1]
```diff
@@ -86,4 +92,12 @@ public void scalingPrerequisites(AuthenticatedContext ac, CloudStack stack, bool
             }
         }
     }
+
+    private boolean checkInstanceStatuses(List<CloudVmInstanceStatus> instanceStatuses, String accountId) {
+        Set<InstanceStatus> allowedStates = EnumSet.of(InstanceStatus.STARTED);
+        if (entitlementService.targetedUpscaleSupported(accountId) && entitlementService.isUnboundEliminationSupported(accountId)) {
```
Comment (horadla23) [inline cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCloudFormationSetup.java:98]: please use `TargetedUpscaleSupportService#targetedUpscaleEntitlementsEnabled`

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsUpscaleService.java [hunk 2]
```diff
@@ -159,14 +163,21 @@ private void validateInstanceStatusesInScaledGroups(AuthenticatedContext ac, Ama
...
+                if (getAllowedInstanceStates(ac.getCloudCredential().getAccountId()).noneMatch(state -> state.equalsIgnoreCase(i.getState().getName()))) {
                     throw new RuntimeException(String.format("Instance (%s) in the group (%s) is not running (%s).",
                             i.getInstanceId(), group.getName(), i.getState().getName()));
                 }
             }
         }
     }
 
+    private Stream<String> getAllowedInstanceStates(String accountId) {
+        if (entitlementService.isUnboundEliminationSupported(accountId) && entitlementService.targetedUpscaleSupported(accountId)) {
```
Comment (horadla23) [inline cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsUpscaleService.java:175]: please use `TargetedUpscaleSupportService#targetedUpscaleEntitlementsEnabled`

**Review by horadla23 — APPROVED** [review]
LGTM

**Review by aszegedi — APPROVED** [review]

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCloudFormationSetup.java [hunk 3]
```diff
@@ -62,7 +68,7 @@ public void scalingPrerequisites(AuthenticatedContext ac, CloudStack stack, bool
             List<CloudInstance> groupInstances = group.getInstances().stream().filter(
                     inst -> InstanceStatus.CREATED.equals(inst.getTemplate().getStatus())).collect(Collectors.toList());
             List<CloudVmInstanceStatus> instanceStatuses = instanceConnector.check(ac, groupInstances);
-            if (!instanceStatuses.stream().allMatch(inst -> inst.getStatus().equals(InstanceStatus.STARTED))) {
+            if (checkInstanceStatuses(instanceStatuses, ac.getCloudCredential().getAccountId())) {
                 String errorMessage = "Not all the existing instances are in [Started] state, upscale is not possible!";
```
[the remark this answers is not in the document]
Comment (bbihari) [reply cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCloudFormationSetup.java]: sure

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**